src/trio/_core/_exceptions.py

```diff
@@ -73,27 +73,30 @@ class Cancelled(BaseException, metaclass=NoPublicConstructor):
 
     """
 
-    source: Literal["deadline", "nursery", "explicit", "unknown", "KeyboardInterrupt"]
-    # TODO: this should probably be a Task?
+    source: Literal[
+        "KeyboardInterrupt", "deadline", "explicit", "nursery", "shutdown", "unknown"
+    ]
```

Comment:
Since we reference this twice, let's pull it out into a global `_CancelSource: TypeAlias = Literal[...]`, which we can also refer to in `_run.py`.

Reply:
not relevant with typing added to `NoPublicConstructor` :rocket:

Reply:
Well, `CancelReason` and `Cancelled` still both use this literal.

Reply:
and I reverted the `NoPublicConstructor` change, so yeah